### DISCUSSION TITLE
メッセージを投稿できる回数制限する？

### DISCUSSION THREAD

**Discussion by Hasuro-R** [Futurealize-with-Tech/fwt#48] [General]
メッセージは1人あたり一つに制限した方が良い？（メンター1人につき一通ずつって意味なので、勿論複数のメンターに送信可能）

無限に投稿できても良いけど、url公開&匿名やから、イタズラが無いとも言い切れない（勿論無いとは思うけど）

万が一めっちゃメッセージ送られて、サーバーに不具合起きて本命が送れんくなるのも嫌やなぁと思ったりもした次第です。

**Comment by xx-n59** [comment]
その方がいいとおもう

実装どうやるか考えてたりする？考えてたら教えて欲しい！考えてなかったら全然大丈夫！

ログイン機能があれば実装すごい楽ではあるんだけど、メンバーの名前で管理するとかは無理だしな〜と思ったから！

**Reply by Hasuro-R** [reply]
個人的に考えてるのは二通りかな

- ローカルストレージで管理（ゆるめ）
- fingerprint-jsを使って管理（厳しい＋不確実）

### ローカルストレージで管理
既にメッセージを送信したメンターの`id`を配列に入れて、それをローカルストレージに保存する。

あとは投稿の時に、既に保存されているidのメンターかどうかを判断して、あったらtoastかなんかで`既に送信しています`みたいに警告すれば良きやと思う。

#### 問題点
- ローカルストレージはユーザーが削除できるので、消したらまた送信可能
- プライベートタブだと使えない

なので制限としては弱め

### fingerprint-jsで管理
`fingerprint-js`ってライブラリがあって、ユーザーのデバイスとかの情報をハッシュ化してユーザー毎に特定の`id`みたいなものをつけられる。

そのハッシュ化した情報をMessageテーブルに一緒に保存して管理する。

これならプライベートタブだろうと、グーグルアカウントを変えようとも管理可能。

#### 問題点
- ユーザーのデバイスなどの多くの情報を元に生成されるので他のユーザーと誤認する可能性がある（勿論ほとんどありえない）
- ブラウザのバージョンをあげたりするだけで認識できなくなるかも

一応手段としてはこんな感じかなぁ

**Reply by xx-n59** [reply]
明日までに考えます！ありがとう！
